sim-lib/src/lib.rs

```diff
@@ -679,15 +693,19 @@ async fn produce_simulation_results(
     loop {
         tokio::select! {
             biased;
-            _ = shutdown.clone() => break,
+            _ = listener.clone() => break,
             output = output_receiver.recv() => {
                 match output{
                     Some(simulation_output) => {
                         match simulation_output{
+                            SimulationOutput::TerminalError(e) => {
```

Comment:
What's the advantage of propagating the `TerminalError` down to this results loop vs exiting in `consume_results`? 

Trying to think about where we'd want this to be handled in the case where we're going to do "smarter" handling of errors (eg, if we wanted to add a feature to allow some activities to die but others continue, this feels like one stage too deep to handle that case?). Difficult to think about it before we've done it :')

Reply:
> What's the advantage of propagating the `TerminalError` down to this results loop vs exiting in `consume_results`?

If you mean exiting in `consume_events` and not `consume_simulation_results`, I think it's quite simple and direct, and will give us the best chance of stopping specific activities on terminal error